src/Microsoft.ML.Data/DataView/DetectAnomalyBySrCnnBatchTransform.cs

```diff
@@ -0,0 +1,333 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Numeric;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Data.DataView
+{
+    public abstract class BatchTransformBase<TInput, TBatch> : IDataView
```

Comment:
Just a copy of Yael's sample for now. Is the intention that this should be a general class somewhere (or only used for the batch scenario)?

Reply:
Responding to the question above and this one together. 
This class should be marked internal and can live here in Microsoft.ML.Data. The specific implementation of DetectAnomalyBySrCnnBatchTransform should move to Microsoft.ML.TimeSeries.

---
In reply to: [424168219](https://github.com/mengaims/machinelearning/pull/1#discussion_r424168219) [](ancestors = 424168219)

Reply:
I agree with Harish. Also, the public API in the ExtensionsCatalog should return an IDataView instead of a DetectAnomalyBySrCnnBatchTransform (which would not be possible anyway once this class is made internal).

---
In reply to: [424170304](https://github.com/mengaims/machinelearning/pull/1#discussion_r424170304) [](ancestors = 424170304,424168219)

Reply:
In response to @klausmh 's question: I think in the future this class might be useful as a base class for other components, for example, @antoniovs1029 mentioned that there may be a need to introduce a batch-scoring version of the ONNX transform.

We should probably also find a better name for it than `BatchTransformBase`, since it is an implementation of `IDataVew` (which doesn't have the word "Transform" in it).


---
In reply to: [424168219](https://github.com/mengaims/machinelearning/pull/1#discussion_r424168219) [](ancestors = 424168219)

Reply:
BatchDataViewMapperBase?

---
In reply to: [424210564](https://github.com/mengaims/machinelearning/pull/1#discussion_r424210564) [](ancestors = 424210564,424168219)

Reply:
@yaeldekel We will no longer introduce a batch version for ONNX transformer. Still seems to be a great idea in case we need something similar in the future.

---
In reply to: [424210564](https://github.com/mengaims/machinelearning/pull/1#discussion_r424210564) [](ancestors = 424210564,424168219)